gtk4/src/expression.rs

```diff
@@ -0,0 +1,362 @@
+    }
+
+    #[doc(alias = "gtk_expression_bind")]
+    pub fn bind<T: IsA<Object>>(
+        &self,
+        target: &T,
+        property_name: &str,
+        this: Option<&T>,
+    ) -> ExpressionWatch {
+        assert_initialized_main_thread!();
+        unsafe {
+            let ret = ffi::gtk_expression_bind(
+                self.to_glib_none().0,
+                target.as_ref().to_glib_none().0,
+                property_name.to_glib_none().0,
+                this.map(|t| t.as_ref()).to_glib_none().0,
+            );
+            ExpressionWatch(*ret)
+        }
+    }
+
+    #[doc(alias = "gtk_expression_evaluate")]
+    pub fn evaluate<T: IsA<Object>>(&self, this: Option<&T>) -> Option<Value> {
+        skip_assert_initialized!();
+        unsafe {
+            let mut value = Value::uninitialized();
+            let ret = ffi::gtk_expression_evaluate(
+                self.to_glib_none().0,
+                this.map(|t| t.as_ref()).to_glib_none().0,
+                value.to_glib_none_mut().0,
+            );
+            if from_glib(ret) {
+                Some(value)
+            } else {
+                None
+            }
+        }
+    }
+
+    #[doc(alias = "gtk_expression_watch")]
+    pub fn watch<T: IsA<Object>, F: Fn() + 'static>(
+        &self,
+        this: Option<&T>,
+        notify: F,
+    ) -> ExpressionWatch {
+        assert_initialized_main_thread!();
+        unsafe extern "C" fn notify_trampoline<F: Fn() + 'static>(user_data: glib::ffi::gpointer) {
+            let f: &F = &*(user_data as *const F);
+            f()
+        }
+        unsafe extern "C" fn destroy_func<F: Fn() + 'static>(user_data: glib::ffi::gpointer) {
+            let _callback: Box_<Option<Box_<F>>> = Box_::from_raw(user_data as *mut _);
+        }
+        let callback_data: Box_<F> = Box_::new(notify);
+        unsafe {
+            let ret = ffi::gtk_expression_watch(
+                self.to_glib_none().0,
+                this.map(|t| t.as_ref()).to_glib_none().0,
+                Some(notify_trampoline::<F> as _),
+                Box_::into_raw(callback_data) as *mut _,
+                Some(destroy_func::<F> as _),
+            );
+            ExpressionWatch(*ret)
+        }
+    }
+}
+
+impl<'a> glib::value::FromValueOptional<'a> for Expression {
+    unsafe fn from_value_optional(value: &'a Value) -> Option<Self> {
+        from_glib_full(ffi::gtk_value_dup_expression(value.to_glib_none().0))
+    }
+}
+
+impl glib::value::SetValue for Expression {
+    unsafe fn set_value(value: &mut Value, this: &Self) {
+        ffi::gtk_value_set_expression(value.to_glib_none_mut().0, this.to_glib_none().0)
+    }
+}
+
+impl glib::value::SetValueOptional for Expression {
+    unsafe fn set_value_optional(value: &mut glib::Value, this: Option<&Self>) {
+        ffi::gtk_value_set_expression(value.to_glib_none_mut().0, this.to_glib_none().0)
+    }
+}
+
+/// Register Expression's ParamSpec support
+trait GtkParamSpecExt {
```

Comment:
There are other pspecs in gtk or not? Maybe move it already to a separate file for that reason :)

Reply:
It's the only pspec gtk has, i guess it's fine to keep it here